### PR TITLE
Part of #3708:  array_api to call functions from arkouda.pdarray_crea…

### PR DESCRIPTION
**Pull request by ajpotts**
Part of #3708: array_api to call functions from arkouda.pdarray_creation

Refactors:

- creation_functions.py
- elementwise_functions.py
- linalg.py

### CODE REVIEW

**Review by stress-tess — APPROVED** [review]
Looks great! thanks amanda!

File: arkouda/array_api/linalg.py [hunk 1]
```diff
@@ -1,36 +1,15 @@
...
     Matrix product of two arrays.
     """
-    from .array_object import Array
-
-    if x1._array.ndim < 2 and x2._array.ndim < 2:
-        raise ValueError(
-            "matmul requires at least one array argument to have more than two dimensions"
-        )
-
-    x1b, x2b, tmp_x1, tmp_x2 = broadcast_if_needed(x1._array, x2._array)
```
Comment (jeremiah-corrado) [inline arkouda/array_api/linalg.py:18]: I don't think ak_matmul does broadcasting. Could you add that there as a part of this PR?

File: arkouda/array_api/linalg.py [hunk 2]
```diff
@@ -44,42 +23,16 @@ def matrix_transpose(x: Array) -> Array:
...
-        },
-    )
-
-    return Array._new(create_pdarray(repMsg))
-
-
-def vecdot(x1: Array, x2: Array, /, *, axis: int = -1) -> Array:
     from .array_object import Array
 
-    x1b, x2b, tmp_x1, tmp_x2 = broadcast_if_needed(x1._array, x2._array)
```
Comment (jeremiah-corrado) [inline arkouda/array_api/linalg.py:67]: same comment here

**Review by jeremiah-corrado — APPROVED** [review]
Looks good, just one minor issue with the linear algebra changes:

File: arkouda/array_api/linalg.py (hunk 1 above)
Comment (ajpotts) [reply arkouda/array_api/linalg.py:18]: I made the change.  Let me know what you think.

File: arkouda/array_api/linalg.py (hunk 2 above)
Comment (ajpotts) [reply arkouda/array_api/linalg.py:67]: This one also.

File: arkouda/array_api/linalg.py (hunk 1 above)
Comment (jeremiah-corrado) [reply arkouda/array_api/linalg.py:18]: Looks good, thanks!

**Review by jeremiah-corrado — APPROVED** [review]

**Review by jaketrookman — APPROVED** [review]

**Comment by ajpotts** [comment]
I had to remove the broadcast from `matmul` to account for this bug: https://github.com/Bears-R-Us/arkouda/issues/3871

**Review by stress-tess — APPROVED** [review]

**Pull request merged**